### PR TITLE
Update docstring for profile_line

### PR DESCRIPTION
**Pull request by sdsingh**
Address docstring clarifications for #1958. 
Please let me know if the doctest looks reasonable.

### CODE REVIEW

File: skimage/measure/profile.py [hunk 1]
```diff
@@ -44,11 +46,18 @@ def profile_line(img, src, dst, linewidth=1,
...
 
     Notes
     -----
     The destination point is included in the profile, in contrast to
     standard numpy indexing.
+
+    For example: 
+        >>> img = np.array([1,1,2,2])
+        >>> profile_line(img, (0, 0), (0, 4)) # The final point is out of bounds
+        >>> profile_line(img, (0, 0), (0, 3)) # This accesses the full first row
```
Comment (jni) [inline skimage/measure/profile.py]: Thanks! All of this should go in the Examples section. Note that doctest allows you to mix code and prose, just separate by blank lines. See: https://docs.python.org/3.5/library/doctest.html

You should add the output and make sure it's correct by running it. Remember that even a single stray character can cause a doctest to fail!

File: skimage/measure/profile.py [hunk 2]
```diff
@@ -16,6 +16,8 @@ def profile_line(img, src, dst, linewidth=1,
         The start point of the scan line.
     dst : 2-tuple of numeric scalar (float or int)
         The end point of the scan line.
+        Note: The final destination point is INCLUDED in the profile, 
+            in constrast to standard numpy indexing.
```
Comment (jni) [inline skimage/measure/profile.py]: Use standard paragraph indentation here. Don't use all-caps. Use `*included*` for emphasis. Remove "Note" and "final".

**Comment by jni** [comment]
@sdsingh thanks for tackling this! I left a few comments but they should be easy to fix!

**Comment by sdsingh** [comment]
No problem! The docstrings run appropriately (after some changes).

File: skimage/measure/profile.py [hunk 3]
```diff
@@ -44,11 +45,16 @@ def profile_line(img, src, dst, linewidth=1,
...
     array([ 1.,  1.,  2.,  2.])
+    >>> profile_line(img, (1, 0), (1, 6), cval=4)
+    array([ 1.,  1.,  1.,  2.,  2.,  2.,  4.])
 
-    Notes
-    -----
     The destination point is included in the profile, in contrast to
     standard numpy indexing.
+    For example: 
+    >>> profile_line(img, (1, 0), (1, 6)) # The final point is out of bounds
```
Comment (stefanv) [inline skimage/measure/profile.py]: One last small and very nitpicky change:  Python PEP8 requires two spaces before the "#" comment character.

**Comment by sdsingh** [comment]
Fixed spacing

**Comment by stefanv** [comment]
:+1: for merging (@sdsingh the project has a policy where two core contributors must agree before merging, so as soon as another one sees this and feels we've addressed all problems, they will click the Big Green Button and merge)

**Comment by jni** [comment]
Thanks @sdsingh!

**Pull request merged**